packages/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -146,11 +146,11 @@ export const GridRootStyles = styled('div', {
   const borderColor = getBorderColor(t);
   const radius = t.shape.borderRadius;
 
-  const containerBackground = t.vars
+  const background = t.vars
     ? t.vars.palette.background.default
-    : (t.mixins.MuiDataGrid?.containerBackground ?? t.palette.background.default);
-
-  const pinnedBackground = t.mixins.MuiDataGrid?.pinnedBackground ?? containerBackground;
+    : (t.mixins.MuiDataGrid?.background ?? t.palette.background.default);
+  const headerBackground = t.mixins.MuiDataGrid?.headerBackground ?? background;
+  const pinnedBackground = t.mixins.MuiDataGrid?.pinnedBackground ?? background;
```

Comment:
Currently, `containerBackground` is applied to column headers and pinned rows. IMO, it makes more sense to use the `pinnedBackground` color for pinned rows. With this change, `containerBackground` would only change the color of column headers, hence why I have renamed it to `headerBackground`.

Any concerns with this change? @mui/xgrid